src/services/exit_order.py

```diff
@@ -0,0 +1,360 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Iterable
+
+from eth_typing import Address
+
+from src.constants import FAR_FUTURE_EPOCH
+from src.modules.accounting.typings import OracleReportLimits
+from src.modules.submodules.typings import ChainConfig
+from src.providers.consensus.typings import Validator
+
+from src.typings import ReferenceBlockStamp
+from src.utils.abi import named_tuple_to_dataclass
+from src.utils.events import get_events_in_past
+from src.web3py.extensions.lido_validators import (
+    LidoValidator,
+    NodeOperator,
+    NodeOperatorGlobalIndex,
+    StakingModuleId,
+    NodeOperatorId,
+    ValidatorsByNodeOperator,
+)
+from src.web3py.typings import Web3
+
+
+@dataclass
+class NodeOperatorPredictableState:
+    predictable_validators_total_age: int
+    predictable_validators_count: int
+    targeted_validators_limit_count: int | None
+    delayed_validators_count: int
+
+
+@dataclass
+class ValidatorToExitIteratorConfig:
+    max_validators_to_exit: int
+    validator_delayed_timeout_in_slots: int
+
+
+class ValidatorToExitIterator:
+    """
+    Exit order predicates sequence:
+    1. Validator whose operator with the lowest number of delayed validators
+    2. Validator whose operator with the highest number of targeted validators to exit
+    3. Validator whose operator with the highest stake weight
+    4. Validator whose operator with the highest number of validators
```

Comment:
maybe it's better to tune up this item (it doesn't map to the actual code rn)

Reply:
will be fixed during refactoring